rest-service/manager_rest/rest/resources_v3_1/plugins.py

```diff
@@ -63,10 +65,31 @@ def post(self, **kwargs):
         """
         Upload a plugin
         """
+
         visibility = rest_utils.get_visibility_parameter(
             optional=True,
             is_argument=True,
             valid_values=VisibilityState.STATES
         )
         with rest_utils.skip_nested_marshalling():
             return super(Plugins, self).post(visibility=visibility)
+
+
+class PluginsId(resources_v2_1.PluginsId):
+
+    @rest_decorators.exceptions_handled
+    @authorize('provider_context_get')
+    @rest_decorators.marshal_with(models.Plugin)
+    def put(self, plugin_id, **kwargs):
+        """
+        Update the plugin's archive_name (remove the `installing-` prefix). '
+        This method is called from the PluginInstaller after the plugin
+        installation has ended.
+        """
+
+        sm = get_storage_manager()
+        plugin = sm.get(models.Plugin, plugin_id)
+        if INSTALLING_PREFIX in plugin.archive_name:
```

Comment:
I would log if the 'else' because this should not happen and this is a source of a bug